CONTRIBUTING.md

```diff
@@ -78,6 +79,56 @@ branches that minor and patch releases will be tagged. In some cases, it may
 be necessary to open PRs for bugfixes directly against stable branches, but
 this should generally not be the case.
 
+### Support and guarantees
```

Comment:
This is language that open source projects need to be very very careful about.

The Metal³ project does not support _anything_, and there are no guarantees. Or, to quote the license:

> Unless required by applicable law or
      agreed to in writing, Licensor provides the Work (and each
      Contributor provides its Contributions) on an "AS IS" BASIS,
      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
      implied, including, without limitation, any warranties or conditions
      of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A
      PARTICULAR PURPOSE.

There can be no suggestion that this is a place we are agreeing (in writing!) to anything more.

Reply:
I think there is confusion about two different kinds of guarantees or warranties here. As per the license, we do not guarantee anything about the "end product". We do not claim that Metal³ is even fit for purpose.
However, that does not mean that we cannot promise to work on Metal³ to make it better. This is about the processes and our work. We can promise to maintain CI coverage and backport fixes to release branches without promising that those fixes or patch releases will be fit for purpose.

If we compare with other projects, I can see that CAPI has the [exact same license](https://github.com/kubernetes-sigs/cluster-api/blob/main/LICENSE) as us and [very similar section on support and guarantees](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#support-and-guarantees) as what is proposed here. Kubernetes also has the same license and still makes promises about [supported versions and version skew](https://kubernetes.io/releases/version-skew-policy/) and [support period](https://kubernetes.io/releases/patch-releases/#support-period).

Based on this I think it is quite safe to have a section like this. Perhaps we can adjust the wording a bit to make it clearer what we mean? Any suggestions?

Reply:
Agree with @lentzi90, the title is influenced by CAPI and two lines down it mentions `Support for this section refers to the ability to backport and release patch versions`. So this is mainly talking about CI support and branch maintenance. Do you have any specific suggestion for wording @zaneb ?

Reply:
> This is the contributor guide, so I'd suggest phrasing it around what sorts of changes we as a project are not prepared to accept in contributions (i.e. breaking changes to the API). 

That section is already there in this document. 

> I would describe the live branches as 'maintained' rather than 'supported'.

I have changed the wording now a bit. PTAL